examples/fragment-performance/test.js

```diff
@@ -0,0 +1,86 @@
+const { analyseHooks } = require('./hooks');
+
+describe("Fragment performance in browser", () => {
+    let server;
+
+    before(() => {
+        const tailor = new Tailor({
+            templatesPath: __dirname + '/templates',
+            pipeAttributes: attributes => {
+                const timingGroups = attributes['timing-group']
+                    ? attributes['timing-group'].split(',')
+                    : [];
+                const { id, primary } = attributes;
+                return { timingGroups, id, primary: !!(primary || primary === '') };
+            },
+            pipeInstanceName: 'TailorPipe',
+            maxAssetLinks: 3
+        });
+        server = http.createServer((req, res) => {
+            if (req.url === '/favicon.ico') {
+                res.writeHead(200, { 'Content-Type': 'image/x-icon' });
+                return res.end('');
+            }
+            return tailor.requestHandler(req, res);
+        });
+        server.listen(8080);
+        tailor.on('error', (request, err) => console.error(err));
+    
+        const fragment1 = http.createServer(
+            serveFragment('Header', 'http://localhost:8081', 1)
+        );
+        fragment1.listen(8081);
+    
+        const fragment2 = http.createServer(
+            serveFragment('Primary', 'http://localhost:8082', 2)
+        );
+        fragment2.listen(8082);
+    
+        const fragment3 = http.createServer(
+            serveFragment('Footer', 'http://localhost:8083', 3, true)
+        );
+        fragment3.listen(8083);
+    })
+
+    after(() => {
+        if(server != null) {
+            server.close();
+        }
+    })
+
+    it("should be able to capture performance metrics in browser", async () => {
+        const browser = await puppeteer.launch();
+        const page = await browser.newPage();
+        try {
+            // Forcing to wait till there are no networking events
+            await page.goto('http://localhost:8080/index', {
+                waitUntil: 'networkidle0'
+            });
+            // Capture all the fragment related metrics
+            const metrics = await page.evaluate(() => {
+                // Serializing the outputs otherwise it will be undefined
+                return [
+                    JSON.stringify(performance.getEntriesByType('mark')),
+                    JSON.stringify(performance.getEntriesByType('measure')),
+                    JSON.stringify(window.TailorPipe.getEntries())
+                ];
+            });
+            const [mark, measure, entries] = [
+                JSON.parse(metrics[0]),
+                JSON.parse(metrics[1]),
+                JSON.parse(metrics[2])
+            ];
+            await analyseHooks(mark, measure, entries);
+            await browser.close();
+        } catch (e) {
+            console.error(e);
+            process.exit(1);
+        }
+    }).timeout(8000); // Add increased timeout since the test runs inside puppeteer
```

Comment:
use timeout from mocha CLI flags and use `done` function since its an async test.

Reply:
We don't need the `done` for an async test if the assertion is done at the right time, which in this case is calling the `analyseHooks` method call and the control is at the end of this function. But since we are not calling any assertion directly in this test, I will add the done for making it explicit

Reply:
The reason is suggested for done is, if we failed to capture any promise rejections and done is never called and we wont be able to see any meaningful error message from mocha.

Reply:
Added

Reply:
On checking the travis failures, it seems we can only use 1 of `async` or `done`. I couldn't recall earlier why I didn't have but now the error logs show the same: https://travis-ci.org/zalando/tailor/jobs/548746746
> 1) Fragment performance in browser should be able to capture performance metrics in browser:
     Error: Resolution method is overspecified. Specify a callback *or* return a Promise; not both.
      at Context.it (examples/fragment-performance/test.js:81:13)

Reply:
Hmmm. interesting, let's remove it then.